File: final_dataset_A/generated_code/chatgpt/java/ProductSearchEndpointDetailed-chatgpt.java
package com.example.demo.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.*;

@RestController
@RequestMapping("/api/products")
public class ProductController {

    @GetMapping("/search")
    public ResponseEntity<Map<String, Object>> searchProducts(
            @RequestParam(required = false) String keyword,
            @RequestParam(required = false) String category,
            @RequestParam(required = false) Double minPrice,
            @RequestParam(required = false) Double maxPrice) {

        // Simulate matching products
        List<Map<String, Object>> products = new ArrayList<>();

        Map<String, Object> product = new LinkedHashMap<>();
        product.put("id", 1);
        product.put("name", "Wireless Mouse");
        product.put("category", "Electronics");
        product.put("price", 29.99);

        products.add(product);

        // JSON response
        Map<String, Object> response = new LinkedHashMap<>();
        response.put("success", true);
        response.put("keyword", keyword);
        response.put("category", category);
        response.put("filters", Map.of(
                "minPrice", minPrice,
                "maxPrice", maxPrice
        ));
        response.put("resultCount", products.size());
        response.put("results", products);

        return ResponseEntity.ok(response);
    }
}

File: generated-code-groupA-updated/generated-code-groupA-updated/chatgpt/python-chatgpt-A/product_search_endpoint_detailed-chatgpt.py
from flask import Flask, request, jsonify

app = Flask(__name__)

@app.route("/products/search", methods=["GET"])
def search_products():
    keyword = request.args.get("keyword", "")

    # Simulated search result
    product = {
        "id": 1,
        "name": "Wireless Mouse",
        "category": "Electronics",
        "price": 29.99
    }

    return jsonify({
        "keyword": keyword,
        "resultCount": 1,
        "results": [product]
    })

if __name__ == "__main__":
    app.run(debug=True)